Updated bug table in README to explain Java 7 -> Java 8 changes

Greg4cr opened the pull request:
This is a minor documentation change to address questions like #299. It notes the valid bug IDs and deprecated bug IDs, along with a short explanation of why some bug IDs are no longer available.

rjust commented:
Can you please also update the [script](https://github.com/rjust/defects4j/blob/additional-projects/framework/util/create_bugs_table.pl) that outputs the table?

Greg4cr commented:
@rjust - Thanks for informing me about that script. I've added it to this branch and updated it to match the style. 

(Bonus - it revealed an error I made in creating the table in the first place)

rjust commented:
@Greg4cr, I made a few minor fixes (renamed the script to match the name in master, and right-align number of bugs).

There is still one issue in the script: for collections, it outputs:
```
Bug IDs in Use
1,25-28

Deprecated Bug IDs (\*)
1-24
```

The `1` in Bug IDs in Use is wrong. (I already fixed this in the README.)
